### PR TITLE
Move to bindgen_cuda.

### PR DESCRIPTION
**Pull request by Narsil**
Moving to `bindgen_cuda` to cleanup the build code.

All the logic should be backward compatible.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -15,8 +15,7 @@ half = { version = "2.3.1", features = ["num-traits"] }
 
 [build-dependencies]
 anyhow = { version = "1", features = ["backtrace"] }
-num_cpus = "1.15.0"
-rayon = "1.7.0"
+bindgen_cuda = { path = "../bindgen_cuda" }
```
Comment (OlivierDehaene) [inline Cargo.toml:18]: Fix?

Comment (Narsil) [inline Cargo.toml:18]:
```suggestion
bindgen_cuda = "0.1.1"
```

**Comment by Narsil** [comment]
> Could bindgen cuda also generate the ffI?

I don't think we should, bindgen already exists for that and is great  (surface is so small here I'm not sure it's worth using something like bindgen).

File: build.rs [hunk 2]
```diff
@@ -41,211 +27,29 @@ fn main() -> Result<()> {
...
-                    .arg(format!("--gpu-architecture=sm_{compute_cap}"))
-                    .arg("-c")
-                    .args(["-o", obj_file.to_str().unwrap()])
-                    .args(["--default-stream", "per-thread"])
+                    .arg("-O3")
                     .arg("-U__CUDA_NO_HALF_OPERATORS__")
                     .arg("-U__CUDA_NO_HALF_CONVERSIONS__")
                     .arg("-U__CUDA_NO_HALF2_OPERATORS__")
                     .arg("-U__CUDA_NO_BFLOAT16_CONVERSIONS__")
+                    .arg("-Icutlass/include")
```
Comment (OlivierDehaene) [inline build.rs:39]: Is this normal?

Comment (Narsil) [reply build.rs:39]: Bad copy paste.

Comment (Narsil) [inline build.rs:39]:
```suggestion
```

**Review by OlivierDehaene — APPROVED** [review]

**Pull request merged**